SignalServiceKit/src/Profiles/OWSProfilesManager.m

```diff
@@ -3,84 +3,100 @@
 //
 
 #import "OWSProfilesManager.h"
+#import "NSData+hexString.h"
+#import "NSDate+OWS.h"
 #import "OWSMessageSender.h"
 #import "SecurityUtils.h"
+#import "TSAccountManager.h"
 #import "TSStorageManager.h"
-#import "TextSecureKitEnv.h"
-
 #import "TSYapDatabaseObject.h"
+#import "TextSecureKitEnv.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-@class TSThread;
+@interface UserProfile : TSYapDatabaseObject
 
-@interface AvatarMetadata : TSYapDatabaseObject
+@property (nonatomic, readonly) NSString *recipientId;
+@property (nonatomic, nullable) NSString *profileName;
+@property (nonatomic, nullable) NSString *avatarUrl;
+@property (nonatomic, nullable) NSString *avatarDigest;
 
 // This filename is relative to OWSProfilesManager.profileAvatarsDirPath.
-@property (nonatomic, readonly) NSString *fileName;
-@property (nonatomic, readonly) NSString *avatarUrl;
-@property (nonatomic, readonly) NSString *avatarDigest;
+@property (nonatomic, nullable) NSString *avatarFileName;
+
+// This should reflect when either:
+//
+// * The last successful update was started.
+// * The in-flight update was started.
+@property (nonatomic, nullable) NSDate *lastUpdateDate;
```

Comment:
Maybe `updateStartedAt` ? otherwise IMO it reads as though update was completed.